Service/AbstractEntityIndexOptionsBuilder.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Algolia\AlgoliaSearch\Service;
+
+use Algolia\AlgoliaSearch\Api\Builder\EntityIndexOptionsBuilderInterface;
+use Algolia\AlgoliaSearch\Api\Data\IndexOptionsInterface;
+use Algolia\AlgoliaSearch\Exceptions\AlgoliaException;
+use Magento\Framework\Exception\NoSuchEntityException;
+
+abstract class AbstractEntityIndexOptionsBuilder extends IndexOptionsBuilder implements EntityIndexOptionsBuilderInterface
+{
+    /**
+     * Safely builds index options with AlgoliaException handling
+     * Template method that handles the try/catch logic for all entity implementations
+     *
+     * @throws NoSuchEntityException
+     */
+    protected function safeBuildWithComputedIndex(string $indexSuffix, int $storeId, bool $isTmp = false): IndexOptionsInterface
```

Comment:
If there a particular reason not to simply replace `buildWithComputedIndex` by this implementation on the parent class ? 
The Index Options management begins to be quite complex and I'm wondering if it's necessary

Reply:
ahah thanks for the clarification 😆